core/src/main/java/io/grpc/ClientCall.java

```diff
@@ -183,4 +183,13 @@ public void onReady() {}
   public boolean isReady() {
     return true;
   }
+
+  /**
+   * Enables per-message compression, if an encoding type has been negotiated.  If no message
+   * encoding has been negotiated, this is a no-op.
```

Comment:
It was in call options, but it isn't granular enough.  Per message compression is needed rather than per call.    

Also, having it in call options would later conflict when per message compression is added in later.  If the call options disagrees with the per message options, which should win?  There would be no such ambiguity if it was settable at the lowest level, which is per message.